src/main/java/org/jenkinsci/plugins/casc/Attribute.java

```diff
@@ -165,12 +168,21 @@ public boolean equals(Owner o1, Owner o2) throws Exception {
         T getValue(O target) throws Exception;
     }
 
-    private Type _getValue(Owner target) throws Exception {
-        final PropertyDescriptor property = PropertyUtils.getPropertyDescriptor(target, name);
-        if (property == null) return null;
-        final Method readMethod = property.getReadMethod();
-        if (readMethod == null) return null;
-        return (Type) readMethod.invoke(target);
+    private Type _getValue(Owner target) throws ConfiguratorException {
+        try {
+            final PropertyDescriptor property = PropertyUtils.getPropertyDescriptor(target, name);
+            if (property != null && property.getReadMethod() != null) {
+                return (Type) property.getReadMethod().invoke(target);
+            }
+            // If this is a public final field, developers don't define getters as jelly can use them as-is
+            final Field field = FieldUtils.getField(target.getClass(), name);
+            if (field == null) {
+                throw new ConfiguratorException("Can't read attribute '" + name + "' from "+ target);
+            }
+            return (Type) field.get(target);
```

Comment:
Still throws RuntimeException

Reply:
maybe I'm missreading (?) but AFAICT it only can throw ConfiguratorException

Reply:
Runtime exceptions are not checked, so `throws` won't help there.
There are several possible runtime exceptions: https://docs.oracle.com/javase/7/docs/api/java/lang/reflect/Field.html#get(java.lang.Object)

Reply:
oh indeed I missed IllegalArgumentException